qemu/tests/virtio_fs_share_data.py

```diff
@@ -393,7 +393,7 @@ def check_socket_group():
                     utils_misc.safe_rmdir(fs_dest, session=session)
     finally:
         if setup_local_nfs:
-            if vm.is_alive():
+            if vm and vm.is_alive():
```

Comment:
The root case is that the `vm ` is not assigned(actual error: `UnboundLocalError: local variable 'vm' referenced before assignment`) before `env_process.preprocess(test, params, env)`, so for your solution, I think it maybe not explain why to do that, I mean there is still a little bit of logical error.
My suggestion that we should assign a default value to `vm`: vm = None
```
        vm = None
        if cmd_xfstest or setup_local_nfs or setup_hugepages:
            params["start_vm"] = "yes"
            env_process.preprocess(test, params, env)
```

Reply:
Yes, `vm` does need to be defined first.
I checked the log and did some tests, the current error is: `local variable 'vm' referenced before assignment`
When the vm creation fails, the vm object does not exist, so we need to set a default value for it in advance.